Src/IronPython/Modules/_io.cs

```diff
@@ -1128,14 +1128,15 @@ public class BufferedWriter : _BufferedIOBase, IDynamicMetaObjectProvider {
             public override BigInteger write(CodeContext/*!*/ context, object buf) {
                 _checkClosed("write to closed file");
 
-                IList<byte> bytes = GetBytes(buf);
+                using var buffer = Converter.Convert<IBufferProtocol>(buf).GetBuffer();
+                var bytes = buffer.AsReadOnlySpan();
                 lock (this) {
                     if (_writeBuf.Count > _bufSize) {
                         FlushNoLock(context);
                     }
 
                     int count = _writeBuf.Count;
-                    _writeBuf.AddRange(bytes);
+                    _writeBuf.AddRange(bytes.ToArray());
```

Comment:
Having to make a copy here is not ideal.

Reply:
I have a bytes enumerator-like extension in the works here for `IPythonBuffer`.

Reply:
Sounds good, will wait for your PR then.

Reply:
OK, I'm gonna jump it to the front of the line.